### PR TITLE
Stratified reports update

### PR DESCRIPTION
**Pull request by camagallen**
## Overview
The data_report feature now accepts y as a data frame, enabling stratified analysis across multiple targets. Also the "overview" column evaluating all features as one unit ("ALL_FEATURES") is now optional via add_overview argument

Resolves: [AB#13809](https://dev.azure.com/kensci/91234890-e1c9-4679-971f-09afd8516db1/_workitems/edit/13809), [AB#15462](https://dev.azure.com/kensci/91234890-e1c9-4679-971f-09afd8516db1/_workitems/edit/15462)

## Checklist
- [X] I have read the [Contributing](https://github.com/KenSciResearch/docs/code_contributions/CONTRIBUTING.md) guidelines
- [X] I performed a self-review of my own code
- [X] The CHANGELOG and version.json have been updated
- [X] Documentation has been updated to reflect the changes
    - [X] The code has clear and helpful comments, particularly in hard-to-understand areas
    - [X] READMEs, Tutorials/Examples, and links have been updated as appropriate
- [X] Tests have been added to prove that the fix is effective or that the feature works
- [X] Pytest was run; new and existing unit tests pass locally with my changes
- [X] I have made note of any breakages or dependencies (if applicable)
- [X] Any dependent changes have been merged and published in downstream modules
- [X] These changes generate no new warnings

## How was this update tested?
pytest and visual inspection of Example-ToolUsage.ipynb

### CODE REVIEW

File: test/test_reports.py [hunk 1]
```diff
@@ -90,13 +90,16 @@ def __dev_test_regression_bias_report(self):
                             pred_type="regression")
 
 
-@pytest.mark.usefixtures("load_classification_data")
+@pytest.mark.usefixtures("load_regression_data")
```
Comment (kenviku) [inline test/test_reports.py:93]: Can you please add the respective work-item for this code?

File: fairmlhealth/reports.py [hunk 2]
```diff
@@ -181,100 +179,131 @@ def color_ratios(row):
...
+            Defaults to None (i.e. all columns).
 
     Requirements:
         Each feature must be discrete to run stratified analysis. If any data
         are not discrete and there are more than 11 values, the reporter will
         reformat those data into quantiles
     """
     #
-    def entr(x):
+    def entrp(x):
```
Comment (kenviku) [inline fairmlhealth/reports.py:216]: nit: rename to `entropy`?

File: test/test_reports.py (hunk 1 above)
Comment (camagallen) [reply test/test_reports.py:93]: @kenviku sorry, what do you mean?

File: fairmlhealth/__preprocessing.py [hunk 3]
```diff
@@ -22,6 +22,9 @@ def prep_X(data):
             X = pd.DataFrame(data, columns=['X'])
     else:
         X = data
+    # Convert columns to numeric type if they do not contain strings
+    for col in X.columns:
```
[the remark this answers is not in the document]
Comment (camagallen) [reply fairmlhealth/__preprocessing.py:26]: @kenviku  We do *not* assume that that every column is numeric. We allow categorical values in the form of strings. The idea here is to force columns to numeric type iff they do not contain strings, hence the use of the errors='ignore' argument in pd.to_numeric.

File: test/test_reports.py (hunk 1 above)
Comment (kenviku) [reply test/test_reports.py:93]: Added the comment at wrong place - it was for the class`TestDataReport`. The comment suggests it to be under development but it was unclear for what or which work-item.

**Review by kenviku — APPROVED** [review]
LGTM

**Pull request merged**